src/Pages/ChannelRequests.razor

```diff
@@ -795,7 +801,12 @@
 
         if (_selectedRequest == null || string.IsNullOrEmpty(_psbtSignRef?.SignedPSBT) || LoggedUser == null)
         {
-            ToastService.ShowError("Error: Not all fields were set");
+            ToastService.ShowError("Not all fields were set");
+        }
+        else if (_selectedRequest.ChannelOperationRequestPsbts.Any(x => x.UserSignerId == LoggedUser?.Id
+                                                                        && _psbtSignRef.SignedPSBT.Equals(x.PSBT)))
+        {
+            ToastService.ShowError("You already signed this PSBT");
```

Comment:
i don't think this will work.
What if for a psbt with 3 needed signers
- user 1 signs, has two tabs open, signs on one of them
- user 2 signs
- user 1 clicks "approve" in second tab, the brought psbt will have the user 2 signature, but will be compared to the old one when user 1 signed.

Also, if you create a user with 2 keys, you can still sign with the same signature twice. I think you'll have to examine the psbt for the xpubs that signed

Reply:
From what I have tested, it works well, what you think is that you get the last signed PSBT when you click approve, but it prints the template:
```
try
{
    var templatePSBT = await BitcoinService.GenerateTemplatePSBT(walletWithdrawalRequest);

    //TODO Save template PSBT (?)
    _templatePsbtString = templatePSBT.ToBase64();
    if (_psbtSignRef != null) await _psbtSignRef.ShowModal();
}
```

Reply:
I just tested, it doesn't let you sign more than the amount of signers, it gives you an error.
But you definitely can sign twice with the same key, and it will stay indefinitely in `Finalizing PSBT`

Reply:
I removed the attribute giving some problems and checked for different users with the same keys. Done 🏁